sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/AggregateCompiler.java

```diff
@@ -339,42 +355,46 @@ void processSingle(SqlSingleValueAggFunction function) {
     void processAvg(SqlAvgAggFunction function) {
         CalciteObject node = CalciteObject.create(function);
         DBSPType aggregatedValueType = this.getAggregatedValueType();
-        DBSPType i64 = new DBSPTypeInteger(CalciteObject.EMPTY, 64, true,true);
+        DBSPType intermediateResultType = aggregatedValueType.setMayBeNull(true);
+        if (aggregatedValueType.is(DBSPTypeInteger.class))
```

Comment:
You like living life on the edge without braces for single line ifs don't you? :)